src/libxrpl/basics/Number.cpp

```
@@ -170,6 +182,61 @@ Number::Guard::round() noexcept
     return 0;
 }
 
+void
+Number::Guard::doRoundUp(rep& mantissa, int& exponent, std::string location)
+{
+    auto r = round();
+    if (r == 1 || (r == 0 && (mantissa & 1) == 1))
```

Should we add enum for `round` rather than use the hardcoded values? It'll be easier to read and less error prone.

---

> Should we add enum for `round` rather than use the hardcoded values? It'll be easier to read and less error prone.

This is another great idea which I think would be better in another PR. Perhaps #6000 or #6025.